examples/solvers/douglas_rachford_primal_dual_heron.py

```diff
@@ -0,0 +1,93 @@
+# Copyright 2014-2016 The ODL development group
+#
+# This file is part of ODL.
+#
+# ODL is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Solves the generalized Heron problem using the Douglas-Rachford solver.
+
+Solves the optimization problem
```

Comment:
Would be good to start with something else than "Solves the". You could say "The generalized Heron problem is defined as"